### PR TITLE
Use the latest Git after compiling it

### LINKED ISSUE #225: Use the latest version of Git
We want to use the latest version of Git for Runners.

https://git-scm.com/book/en/v2/Getting-Started-Installing-Git

### PR DESCRIPTION
**Pull request by yykamei**
git(1) installed via apt-get(8) is a bit old, but we want to use the
latest version of Git. This change compiles Git manually and install it
in the images.

Close #225

### CODE REVIEW

File: base/Dockerfile.git.erb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      ;
+RUN curl -fsSLO --compressed "${GIT_SOURCE_URL}" && \
+    curl -fsSL --compressed "${GIT_SOURCE_SHA256_URL}" | \
+      grep "${GIT_SOURCE_TARBALL}" | \
+      sha256sum --check --strict && \
+    mkdir -p /tmp/git-build && tar -xf "${GIT_SOURCE_TARBALL}" -C /tmp/git-build --strip-components=1
+RUN cd /tmp/git-build && \
+    make configure && \
+    ./configure && \
+    make install DESTDIR=/opt/git
```
Comment (yykamei) [inline base/Dockerfile.git.erb:28]: I used the multi-stage build, so I think the cleanup processes are not necessary, such as `rm -rf /var/lib/apt/lists/*`.

File: base/Dockerfile [hunk 2]
```diff
@@ -2,6 +2,36 @@
...
+    apt-get install -qq -y --no-install-recommends \
+      dh-autoreconf \
+      libcurl4-gnutls-dev \
+      libexpat1-dev \
+      gettext \
+      libz-dev \
+      libssl-dev \
+      curl \
+      ca-certificates \
+      ;
```
Comment (yykamei) [inline base/Dockerfile:23]: I followed [this](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git) instruction and set the packages.

File: base/Dockerfile.git.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+FROM debian:buster-20200607 AS git-builder
+
+ARG GIT_VERSION=<%= ENV.fetch('GIT_VERSION') %>
+ARG GIT_SOURCE_TARBALL=git-${GIT_VERSION}.tar.xz
+ARG GIT_SOURCE_URL=https://mirrors.edge.kernel.org/pub/software/scm/git/${GIT_SOURCE_TARBALL}
+ARG GIT_SOURCE_SHA256_URL=https://mirrors.edge.kernel.org/pub/software/scm/git/sha256sums.asc
+ARG DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -y && \
```
Comment (ybiquitous) [inline base/Dockerfile.git.erb:9]: [suggestion] How about adding a link to the document?

```suggestion
# See https://git-scm.com/book/en/v2/Getting-Started-Installing-Git#_installing_from_source
RUN apt-get update -y && \

File: base/Dockerfile.git.erb (hunk 1 above)
Comment (ybiquitous) [reply base/Dockerfile.git.erb:28]: [question] Why using 2 `RUN`? Does it make sense to unify them to 1 `RUN`?

File: base/Dockerfile.prepare.erb [hunk 4]
```diff
@@ -1,10 +1,15 @@
 COPY base/bin/prepare /tmp/devon_rex_work/
 RUN /tmp/devon_rex_work/prepare
 
+# Install Git and verify the specified version of Git is installed
+COPY --from=git-builder /opt/git /
```
Comment (ybiquitous) [inline base/Dockerfile.prepare.erb:5]: [suggestion] How about copying to `/opt/git` as follows?

```suggestion
COPY --from=git-builder /opt/git /opt/git/
```

Comment (yykamei) [reply base/Dockerfile.prepare.erb:5]: If I changed it like that, I have to reconfigure `PATH` with `PATH=/opt/git/usr/local/bin:$PATH`. I think git(1) should be located in `/usr/local/bin`.

File: base/Dockerfile.git.erb (hunk 3 above)
Comment (yykamei) [reply base/Dockerfile.git.erb:9]: :clap: Thank you.

File: base/Dockerfile.prepare.erb (hunk 4 above)
Comment (ybiquitous) [reply base/Dockerfile.prepare.erb:5]: Oh, I got it. 👍

File: base/Dockerfile.git.erb (hunk 1 above)
Comment (ybiquitous) [reply base/Dockerfile.git.erb:29]: Make sense. 👍

**Comment by yykamei** [comment]
Thank you

**Pull request merged**